opensearch_py_ml/ml_models/sparse_encoding_model.py

```diff
@@ -0,0 +1,332 @@
+
+        self.torch_script_zip_file_path = zip_file_path
+        print("zip file is saved to ", zip_file_path, "\n")
+        return zip_file_path
+
+    def make_model_config_json(
+        self,
+        model_name: str = None,
+        version_number: str = "1.0.0",
+        model_format: str = "TORCH_SCRIPT",
+        description: str = None,
+        model_zip_file_path: str = None,
+    ) -> str:
+        folder_path = self.folder_path
+        if model_name is None:
+            model_name = self.model_id
+
+        model_config_content = {
+            "name": model_name,
+            "version": version_number,
+            "model_format": model_format,
+            "function_name": SPARSE_ENCODING_FUNCTION_NAME,
+        }
+        if model_zip_file_path is None:
+            model_zip_file_path = (
+                self.torch_script_zip_file_path
+                if model_format == "TORCH_SCRIPT"
+                else self.onnx_zip_file_path
+            )
+            if model_zip_file_path is None:
+                raise Exception(
+                    "The model configuration JSON file currently lacks the 'model_content_size_in_bytes' and "
+                    "'model_content_hash_value' fields. You can include these fields by specifying the "
+                    "'model_zip_file_path' parameter. Failure to do so may result in the model registration process "
+                    "encountering issues."
+                )
+            else:
+                model_config_content["model_content_size_in_bytes"] = os.stat(
+                    model_zip_file_path
+                ).st_size
+                model_config_content["model_content_hash_value"] = (
+                    _generate_model_content_hash_value(model_zip_file_path)
+                )
+        if description is not None:
+            model_config_content["description"] = description
+
+        model_config_file_path = os.path.join(
+            folder_path, "ml-commons_model_config.json"
+        )
+        os.makedirs(os.path.dirname(model_config_file_path), exist_ok=True)
+        with open(model_config_file_path, "w") as file:
+            json.dump(model_config_content, file, indent=4)
+        print(
+            "ml-commons_model_config.json file is saved at : ", model_config_file_path
+        )
+        return model_config_file_path
+
+    def get_backbone_model(self):
+        if self.backbone_model is not None:
+            return self.backbone_model
+        else:
+            return AutoModelForMaskedLM.from_pretrained(self.model_id)
+
+    def get_model(self):
+        return NeuralSparseModel(self.get_backbone_model(), self.get_tokenizer())
+
+    def save(self, path):
+        backbone_model = self.get_backbone_model()
+        tokenizer = self.get_tokenizer()
+        backbone_model.save_pretrained(path)
+        tokenizer.save_pretrained(path)
+
+    def post_process(self):
+        pass
+
+    def pre_process(self):
+        pass
```

Comment:
Not seeing usage for these two methods. Are they used by child classes for concrete sparse model?